tailcfg/derpmap.go

```diff
@@ -92,6 +92,9 @@ type DERPRegion struct {
 	// "San Francisco", "Singapore", "Frankfurt", etc.
 	RegionName string
 
+	// Location optionally describes where the DERP region is located.
+	Location Location `json:",omitempty"`
```

Comment:
omitempty doesn't work on structs. Use a pointer here.

I'm not a huge fan of using Location here, though, because now there are two redundant RegionCode/RegionName fields and it's ambiguous which takes priority.

You're only doing this for Lat/Long.

Maybe we need a type for just Lat/Long instead of all the human strings/codes.

Reply:
that's a good point - switching back to lat/long for derp region

Reply:
Is it actually redundant though? The RegionCode/RegionName things make no promises about having any particular value or format, they're for our convenience. Location is more "this is an airport code" / "this is a city name" and can be used for looking up a location.

Anyway if you'd rather just have lat/long in here that's okay with me.

Reply:
Admittedly it might be nice to have e.g. country code in the DERPRegion, so Location gives us that.

But that still doesn't give us "is it in the EU?" for instance, which is a class of request we get, so we'd still need server-side DERPMap massaging to make a compliant map. It's not something we'd need a client to do.

I'm fine having a `*Location` if it's documented which fields are relevant and which take priority if there's any ambiguity. e.g. RegionName vs Location.City.

Reply:
I don't really feel that strongly, there isn't an immediate plan for using the extra detail and extra protocol fields that are unused can make things annoying later. Let's just leave it how it is now rather than thrashing back and forth. We can change it later if we need to.